src/Window.cpp

```diff
@@ -1244,23 +1362,56 @@ void Window::processQueuedCommands() {
 
 double Window::progressRequired(Movement movement) {
     switch (movement) {
-        case Movement::MOVE_FORWARD:
+        case Movement::MOVE_FULLFORWARD:
             return PROGRESS_REQUIRED_FOR_MOVE;
+        case Movement::MOVE_HALFFORWARD:
+        return PROGRESS_REQUIRED_FOR_HALFMOVE;
+        case Movement::MOVE_EDGEFORWARD:
+        return PROGRESS_REQUIRED_FOR_EDGEMOVE;
+        case Movement::TURN_RIGHT45:
+        case Movement::TURN_LEFT45:
+        return PROGRESS_REQUIRED_FOR_HALFTURN;
+        case Movement::RUN_RIGHT:
+        case Movement::RUN_LEFT:
+        return PROGRESS_REQUIRED_FOR_RUNTURN;
         case Movement::TURN_RIGHT:
         case Movement::TURN_LEFT:
             return PROGRESS_REQUIRED_FOR_TURN;
+        case Movement::TURN_BACK:
+        return PROGRESS_REQUIRED_FOR_BACKTURN;
         default:
-            ASSERT_NEVER_RUNS();
-    }
+        stopRun("Stop running because of an exceptional movement!");
+        //ASSERT_NEVER_RUNS();
+        return 0.0f;
```

Comment:
Same comment as above regarding the `ASSERT_NEVER_RUNS` branch. Why are we hitting this branch now?